playbooks/roles/forum_docker/defaults/main.yml

```diff
@@ -0,0 +1,105 @@
+---
+
+forum_docker_image: "edxops/forum"
+forum_docker_tag: "hawthorn.master"
```

Comment:
Any particular reason that we would want to use the images from Tutor rather than directly from upstream? Especially now that upstream is moving in a more docker-friendly direction? I'm guessing that Tutor's docker images are constructed in a more efficient, standard way, but since we're just pulling it and not building or modifying it, it seems simpler to just stick with upstream.

`edxops/forum` is what we currently use in devstack, so I figured it would be good to have exactly the same image there as gets run in prod.

Either way, I think this puts us in a position where we can actually choose between different options without too much trouble.

Reply:
@thraxil yes I'm not sure if `edxops/forum` is designed for production use at all. I'm a bit behind on the edX dockerization efforts. The Overhang images should be a bit better on that regard, at least that's what the advertised value of Tutor: being able to run Dockerized Open edX in production.

I think it's good to see if the edX team have posted any warnings against using those images on production e.g. having debug enabled in the form of an explicit application debugging or just too verbose logging.

Reply:
Cool. Yeah, that's a reasonable concern.

From poking around inside the `edxops/forum` image, it *looks* like it was constructed essentially by running the same ansible roles that we would normally use to do the prod deploy. That makes for a larger image than the Tutor version and is probably slower to build. But those aren't really concerns for us at the moment. Importantly though, the `config/unicorn.rb` and `config/unicorn_tcp.rb` config files which define how the forum process runs in prod are identical to what we already use, so I'd expect the production behaviour to be the same.

The Tutor [Dockerfile](https://github.com/overhangio/tutor/blob/master/tutor/templates/build/forum/Dockerfile) looks pretty solid in comparison (though Juniper so we probably can't just use it). It builds things in a straightforward way and has more of a "proper" docker setup with explicity environment variables, port, and entrypoint configs. It puts things in different locations though (`/openedx` instead of `/edx` so we'd have to change the config templates and test to make sure we did it correctly).

So, I think the `edxops` image is a good starting point for us and a "bridge" to begin converting things to docker while keeping as much as possible compatible with the old version.

Reply:
Cool. That should be a great starting point for us.